### PR TITLE
XW-3998 | Moving mobile-wiki to k8s

### PR DESCRIPTION
**Pull request by xkxd**
## Links

https://wikia-inc.atlassian.net/browse/XW-2978 + subtickets - main ticket
https://wikia-inc.atlassian.net/browse/XW-3998 - dockerfile related ticket

## Description
- dockerfiles for dev and prod
- k8s yaml descriptor templates
- bash scripts for jenkins
- jenkinsfile
- automated by Jenkins Jobs Builder

🎉 

## Reviewers
@Wikia/x-wing

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ module.exports = function (environment) {
 			// when it is created
 		},
 		fastboot: {
-			hostWhitelist: [/.*\.wikia-dev\.(pl|us)/, /.*\.wikia-staging.com/, /.*\.wikia.com/],
+			hostWhitelist: [/.*\.wikia-dev\.(pl|us)/, /.*\.wikia-staging.com/, /.*\.wikia.com/,  /^localhost:\d+$/],
```
Comment (xkxd) [inline config/environment.js]: it's related to an error occurring while trying to run docker image locally, not sure if we want it now (or ever)

File: Dockerfile [hunk 2]
```diff
@@ -14,8 +14,54 @@ RUN useradd -u 663 release
...
+# store prod dependencies separately for docker's caching reasons
+RUN mkdir prod_dependencies
+RUN cp -R node_modules prod_dependencies
+RUN cp -R bower_components prod_dependencies
+
+# install all dependencies
+RUN npm run setup
+
+# copy app
+COPY . .
```
Comment (bkoval) [inline Dockerfile]: Could we add a comment explaining the dots?

**Review by bkoval — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -14,8 +14,60 @@ RUN useradd -u 663 release
...
-COPY . /build
-WORKDIR /build
+# create folder to cache dependencies
+WORKDIR /app
+COPY package.json .
+COPY bower.json .
+
+# install npm globals
+RUN npm install -g bower
+RUN npm install -g ember-cli
```
Comment (rogatty) [inline Dockerfile]: These two shouldn't be required as we don't run them directly but through npm scripts which use the local version anyway.

File: Dockerfile [hunk 4]
```diff
@@ -14,8 +14,60 @@ RUN useradd -u 663 release
...
+RUN npm run clean-deep
```
Comment (slayful) [inline Dockerfile]: Is this necessary?

Comment (xkxd) [reply Dockerfile]: no, it's not

File: Dockerfile.build [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        git
+
+# create folder to cache dependencies
+WORKDIR /app
+COPY package.json .
+COPY bower.json .
+
+# install npm globals
+RUN npm install -g bower
+RUN npm install -g ember-cli
```
Comment (rogatty) [inline Dockerfile.build]: How about using `npm run setup-for-local`? We should lock the versions and it's done there.

File: tasks/docker-build.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+docker cp temporary-container:/app/config  ./docker_temp_container
+docker cp temporary-container:/app/lib  ./docker_temp_container
+docker cp temporary-container:/app/package.json  ./docker_temp_container
+docker cp temporary-container:/app/bower.json  ./docker_temp_container
+
+docker cp temporary-container:/app/prod_dependencies/node_modules ./docker_temp_container
+docker cp temporary-container:/app/prod_dependencies/bower_components ./docker_temp_container
+
+# temporary container is not needed anymore so we're removing it
+docker rm -f temporary-containergit
```
Comment (slayful) [inline tasks/docker-build.sh]: Is that correct?

File: Dockerfile.build [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+# - second argument is the docker daemon (newly created container)
+# - "." means "current working directory", and since we're not selecting any specific files
+# it's basically "copy everything from mobile-wiki project to WORKDIR in docker"
+COPY . .
+
+# build app
+RUN npm run build-prod
+
+RUN apk del .gyp python make g++ git
+RUN npm uninstall -g bower
```
Comment (rogatty) [inline Dockerfile.build]: According to https://www.ctl.io/developers/blog/post/optimizing-docker-images/#image-size it doesn't make sense as the final image size won't shrink.

File: Dockerfile.build (hunk 5 above)
Comment (xkxd) [reply Dockerfile.build]: we don't have it in `mobile-wiki` but thanks, worth considering :)

File: app/services/logger.js [hunk 8]
```diff
@@ -74,17 +74,16 @@ export default Service.extend({
 
 	setupBunyan() {
 		const bunyan = FastBoot.require('bunyan');
-		const BunyanSyslog = FastBoot.require('bunyan-syslog');
+		const process = FastBoot.require('process');
+
 		const instance = bunyan.createLogger({
 			appname: 'mobile-wiki',
 			name: 'mobile-wiki',
+			loggerName: 'services/logger.je',
```
Comment (slayful) [inline app/services/logger.js]: perhaps `logger.js`?

File: k8s/k8s-descriptor-template-preview.yaml [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        env:
+          - name: WIKIA_ENVIRONMENT
+            value: prod
+          - name: WIKIA_DATACENTER
+            value: sjc
+          - name: MEDIAWIKI_DOMAIN
+            value: prod.icache.service.sjc.consul
+          - name: GA_USERID_SALT
+            valueFrom:
+              secretKeyRef:
```
Comment (rogatty) [inline k8s/k8s-descriptor-template-preview.yaml]: We should use https://github.com/Wikia/vault-init-container instead.

**Pull request merged**